packages/router/src/grpc/view-protocol-server/transaction-planner.ts

```diff
@@ -1,78 +1,25 @@
 import type { Impl } from '.';
 import { servicesCtx } from '../../ctx';
-import { getAddressByIndex, TxPlanner } from '@penumbra-zone/wasm';
-import { AddressIndex } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/keys/v1/keys_pb';
+import { planTransaction } from '@penumbra-zone/wasm';
 import { Code, ConnectError } from '@connectrpc/connect';
-import { gasPrices } from './gas-prices';
-import { GasPrices } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/component/fee/v1/fee_pb';
-import { getAssetMetadata } from './helpers';
 
 export const transactionPlanner: Impl['transactionPlanner'] = async (req, ctx) => {
   const services = ctx.values.get(servicesCtx);
   const {
     indexedDb,
     viewServer: { fullViewingKey },
-    querier,
   } = await services.getWalletServices();
+
   const fmdParams = await indexedDb.getFmdParams();
   if (!fmdParams) throw new ConnectError('FmdParameters not available', Code.FailedPrecondition);
   const { chainId, sctParams } = (await indexedDb.getAppParams()) ?? {};
   if (!sctParams) throw new ConnectError('SctParameters not available', Code.FailedPrecondition);
   if (!chainId) throw new ConnectError('ChainId not available', Code.FailedPrecondition);
+  const gasPrices = await indexedDb.getGasPrices();
+  if (!gasPrices) throw new ConnectError('Gas prices is not available', Code.FailedPrecondition);
```

Comment:
It probably makes sense to keep these checks even though wasm will read this data directly from indexed-db. This way we can return an error with FailedPrecondition code to clients